src/crimson/common/errorator.h

```diff
@@ -1007,6 +1046,53 @@ struct errorator {
     return e.error_t<std::decay_t<ErrorT>>::to_exception_ptr();
   }
 
+  class parallel_for_each_state final : private seastar::continuation_base<> {
+    using future_t = errorator<AllowedErrors...>::future<>;
+    std::vector<future_t> _incomplete;
+    seastar::promise<> _result;
+    std::exception_ptr _ex;
+  private:
+    void wait_for_one() noexcept {
+      while (!_incomplete.empty() && _incomplete.back().available()) {
+	if (_incomplete.back().failed()) {
+	  _ex = _incomplete.back().get_exception();
+	}
+	_incomplete.pop_back();
+      }
+      if (!_incomplete.empty()) {
+	seastar::internal::set_callback(_incomplete.back(), static_cast<continuation_base<>*>(this));
+	_incomplete.pop_back();
+	return;
+      }
+      if (__builtin_expect(bool(_ex), false)) {
+	_result.set_exception(std::move(_ex));
+      } else {
```

Comment:
nit: mixing tabs and spaces makes the code a little harder to read while reviewing via GH.